internal/provider/kubernetes/controller_offline.go

```diff
@@ -75,6 +75,21 @@ func NewOfflineGatewayAPIController(
 		envoyGateway:      cfg.EnvoyGateway,
 		mergeGateways:     sets.New[string](),
 		extServerPolicies: extServerPoliciesGVKs,
+		// We assume all CRDs are available in offline mode.
+		backendCRDExists:       true,
+		bTLSPolicyCRDExists:    true,
+		btpCRDExists:           true,
+		ctpCRDExists:           true,
+		eepCRDExists:           true,
+		epCRDExists:            true,
+		eppCRDExists:           true,
+		hrfCRDExists:           true,
+		grpcRouteCRDExists:     true,
+		serviceImportCRDExists: true,
+		spCRDExists:            true,
+		tcpRouteCRDExists:      true,
+		tlsRouteCRDExists:      true,
+		udpRouteCRDExists:      true,
```

Comment:
this is set in the resource watch cycle in the default k8s controller, so without this, they are all false in offline mode.